angel_system/eval/visualization.py

```diff
@@ -151,3 +155,78 @@ def plot_pr_curve(self, detect_intersection_thr=0.1):
         ax.set_ylabel("Precision")
 
         fig.savefig(f"{self.output_dir}/PR.png")
+
+    def plot_roc_curve(self, detect_intersection_thr=0.1):
+        # ============================
+        # Setup figure
+        # ============================
+        fig, ax = plt.subplots(figsize=(7, 8))
+
+        ax.set_xlim([0.0, 1.0])
+        ax.set_ylim([0.0, 1.05])
+        ax.set_title("Receiver Operating Characteristic (ROC)")
+        ax.set_xlabel("False Positive Rate")
+        ax.set_ylabel("True Positive Rate")
+
+        colors = plt.cm.rainbow(np.linspace(0, 1, len(self.labels)))
+
+        # ============================
+        # Get ROC and AUC per class 
+        # ============================
+        fpr = dict()
+        tpr = dict()
+        roc_auc = dict()
+        for i, row in self.labels.iterrows():
+            id = row['id']
+            label = row['class']
+
+            det_ranges = self.dets.loc[self.dets['class'] == label]
+
+            truth = [1 if det['detect_intersection'] > detect_intersection_thr else 0 for i, det in det_ranges.iterrows()]
+            pred = [det['conf'] for i, det in det_ranges.iterrows()]
+
+            fpr[i], tpr[i], _ = roc_curve(truth, pred)
+            roc_auc[i] = auc(fpr[i], tpr[i])
+
+            ax.plot(
+                fpr[i],
+                tpr[i],
+                color=colors[i],
+                lw=2,
+                label="ROC curve of class {0} (area = {1:0.2f})".format(i, roc_auc[i]),
```

Comment:
I just noticed this was the case for the PR plot output as well, but is it possible/sensible to label the semantic class name here instead of just an integer? This is edging towards bikeshedding, which I don't want to enter, but I think including the semantic label here will help with understanding what the plot is showing without going to external resources (e.g. needing to maintain a specific label file with plots being output, putting plots into slides, etc.).